Documentation on using fetch-forms-from-google-drive (#331)
medic/cht-docs: content/en/apps/guides/forms/google-docs.md

---
title: "Fetching forms from Google Docs"
linkTitle: "Fetching forms from Google Docs"
weight: 
description: >
  Using Medic-Conf to fetch files from Google Docs
relatedContent: >
  apps/guides/forms/
---

This action requires two files in the folder you are going to execute this command from. The files are `forms-on-google-drive.json` and  `.gdrive.secrets.json`. 

## forms-on-google-drive.json 

This file should be a json file with a key value pair. The key being the folder/name of the doc when it gets saved locally. The value being the ID from the document saved in google drive. 

> To get the ID for the form. Open the form in google sheets. The URL will look like this https://docs.google.com/spreadsheets/d/12345ABCDEF/edit#gid=555666888 . The ID is the middle portion `12345ABCDEF`

Reviewer: We should make sure to be consistent with capitalization of proper nouns.

```suggestion
To get the File ID for the form open the form in Google Sheets. The URL will look something like `https://docs.google.com/spreadsheets/d/12345ABCDEF/edit#gid=555666888` . The ID is the middle portion `12345ABCDEF`
```

```json
{
  "app/sample_report.xlsx": "12345ABCDEF",
  "contact/person-create.xlsx": "ABCDEF123456",
}
```

## .gdrive.secrets.json

> This file contains authentication secrets that are generated by enabling access to your drive using these [instructions](https://developers.google.com/drive/api/v3/about-auth)

Reviewer: ```suggestion
This file contains authentication secrets that are generated by enabling access to your Google Drive using these [instructions](https://developers.google.com/drive/api/v3/about-auth).
```

```json
{
		"client_id": "your_client_id",
		"project_id": "your_project_id",
		"auth_uri": "https://accounts.google.com/o/oauth2/auth",
		"token_uri": "https://accounts.google.com/o/oauth2/token",
		"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
		"client_secret": "your_client_secret",
		"redirect_uris": ["your_redirect_uris", "http://localhost"]
}
```

> ## Run the Medic Conf Action
> 
> With the files in place. Run `medic-conf fetch-forms-from-google-drive`
> 
> Medic-conf will then prompt for an access code from your authenticated user. Follow the prompts through google's authentication and copy the access code into your command line. The forms will then be copied to the forms folder in the directory you are executing from. 
> 
> *Note: If the browser does not launch own its own. Copy the URL and paste the authentication url into a browser and then follow the prompts.*

Reviewer: ```suggestion
## Run

With the files in place, run medic-conf's `fetch-forms-from-google-drive` action in your CHT app's top level directory.

```sh
medic-conf fetch-forms-from-google-drive.
```

You will be prompted for an access code from your authenticated user. Follow the prompts through Google's authentication page and copy the access code into your command line. The forms will then be downloaded to the locations set in the `forms-on-google-drive.json` file. 

{{% alert title="Note" %}} 
If the browser does not launch own its own copy the URL and paste the authentication url into a browser and then follow the prompts.
{{% /alert %}}
```
